contracts/exchange/UniswapExchange.sol

```diff
@@ -88,10 +88,7 @@ contract UniswapExchange {
         invariant = ethPool.mul(tokenPool);
         shares[msg.sender] = 1000;
         totalShares = 1000;
-        require(
```

Comment:
I see that you have removed all these `require`.

I'm a bit confused by them myself, but I think they might arise from the fact that even if the tokens exchanged conform to IERC20.sol, there is no guarantee that the underlying implementation will be the OpenZeppelin one (which throws on failure).

You used SafeERC20 a while ago, I think that this is the context where it actually becomes useful.

I'm not sure of the way forward. Maybe the requires should be removed from UniswapExchange.sol, but UniswapFactory.sol should wrap the tokens in a SafeERC20.sol contract. What do you think?

Reply:
I think everything is fine. Our contract source code uses the OpenZeppelin implementation, after all, one implementation or another of the standard has to be used in practice, and OpenZeppelin is our offer. If the client wants to change the OpenZeppelin implementation, then they are free to also deal with the implications of using another package-deal.

Reply:
I added a README and put a comment to the effect. If this would be a contract for reuse then we should account for different ERC20 implementations, but since it's main purpose is to show how to implement this type of market I'm inclined for the simpler approach.